src/librustc_mir/transform/mod.rs

```diff
@@ -434,6 +434,9 @@ fn run_optimization_passes<'tcx>(
     promoted: Option<Promoted>,
 ) {
     let mir_opt_level = tcx.sess.opts.debugging_opts.mir_opt_level;
+
+    // Optimizations that need to be done before lowering generators and the generator lowering
```

Comment:
```suggestion
    // Optimizations that need to be done before lowering generators, and the generator lowering
```
Otherwise this parses to me as "before (lowing generator and the generator lowering)"